packages/shadcn/src/SelectWidget/SelectWidget.tsx

```diff
@@ -0,0 +1,93 @@
+import {
+  ariaDescribedByIds,
+  enumOptionsIndexForValue,
+  enumOptionsValueForIndex,
+  FormContextType,
+  RJSFSchema,
+  StrictRJSFSchema,
+  WidgetProps,
+} from '@rjsf/utils';
+
+import { FancyMultiSelect } from '../components/ui/fancy-multi-select';
+import { FancySelect } from '../components/ui/fancy-select';
+import { cn } from '../lib/utils';
+
+/** The `SelectWidget` is a widget for rendering dropdowns.
+ *  It is typically used with string properties constrained with enum options.
+ *
+ * @param props - The `WidgetProps` for this component
+ */
+export default function SelectWidget<
+  T = any,
+  S extends StrictRJSFSchema = RJSFSchema,
+  F extends FormContextType = any
+>({
+  id,
+  options,
+  required,
+  disabled,
+  readonly,
+  value,
+  multiple,
+  autofocus,
+  onChange,
+  onBlur,
+  onFocus,
+  defaultValue,
+  placeholder,
+  rawErrors = [],
+}: WidgetProps<T, S, F>) {
+  const { enumOptions, enumDisabled, emptyValue: optEmptyValue } = options;
+
+  const _onFancyFocus = () => {
+    onFocus(id, enumOptionsValueForIndex<S>(value, enumOptions, optEmptyValue));
+  };
+
+  const _onFancyBlur = () => {
+    onBlur(id, enumOptionsValueForIndex<S>(value, enumOptions, optEmptyValue));
+  };
+
+  const items = (enumOptions as any)?.map(({ value, label }: any, index: number) => ({
+    value: multiple ? value : index.toString(),
+    label: label,
+    index,
+    disabled: Array.isArray(enumDisabled) && enumDisabled.includes(value),
+  }));
+
+  return !multiple ? (
+    <div className='p-0.5'>
```

Comment:
Given that this is the same wrapper div as below, consider rendering just one of them with the ternary inside of it. Something like:
```
   return (
      <div className='p-0.5'>
        {!multiple ? <FancySelect .../> : <FancyMultiSelect .../>}
      </div>
   );
```